src/qtism/data/BranchRuleTargetException.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * Copyright (c) 2013-2015 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ * @author Jérôme Bogaerts <jerome@taotesting.com>
+ * @license GPLv2
+ */
+
+namespace qtism\data;
+
+use \Exception;
+
+/**
+ * An exception to be thrown when an error occurs while dealing with
+ * target of BranchingRules in AssessmentTest.
+ *
+ * @author Tom Verhoof <tomv@taotesting.com>
+ *
+ */
+class BranchRuleTargetException extends Exception
+{
```

Comment:
I would suggest the list of the following exception codes (as class constants) that will help client code to diagnose issues.

UNKNOWN_TARGET = 0
RECURSIVE_BRANCHING = 1
BACKWARD_BRANCHING = 2

Please see my comment in your code throwing this exception for usage.